src/datasets/features/features.py

```diff
@@ -1208,6 +1208,20 @@ def __setitem__(self, column_name: str, feature: FeatureType):
         super().__setitem__(column_name, feature)
         self._column_requires_decoding[column_name] = require_decoding(feature)
 
+    def __delitem__(self, column_name: str):
+        super().__delitem__(column_name)
+        del self._column_requires_decoding[column_name]
+
+    def update(self, iterable, **kwds):
+        if hasattr(iterable, "keys"):
+            for key in iterable.keys():
+                self[key] = iterable[key]
+        else:
+            for key, value in iterable:
+                self[key] = value
+        for key in kwds:
+            self[key] = kwds[key]
+
```

Comment:
@albertvillanova @lhoestq This is a temporary fix for `Features.flatten`. After we merge this one, I'll open a new PR to fix the issues with `dict` subclassing.